src/mercat/mod.rs

```diff
@@ -408,66 +266,11 @@ pub struct AssetTxContent {
     pub auditors_payload: Vec<AuditorPayload>,
 }
 
-#[derive(Clone)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[cfg_attr(feature = "std", derive(Debug))]
-pub struct InitializedAssetTx {
-    pub content: AssetTxContent,
-    pub sig: Signature,
-}
-
-impl Encode for InitializedAssetTx {
-    #[inline]
-    fn size_hint(&self) -> usize {
-        self.content.size_hint() + schnorrkel::SIGNATURE_LENGTH
-    }
-
-    fn encode_to<W: Output>(&self, dest: &mut W) {
-        self.content.encode_to(dest);
-        self.sig.to_bytes().encode_to(dest);
-    }
-}
-
-impl Decode for InitializedAssetTx {
-    fn decode<I: Input>(input: &mut I) -> Result<Self, CodecError> {
-        let content = <AssetTxContent>::decode(input)?;
-        let sig = <[u8; schnorrkel::SIGNATURE_LENGTH]>::decode(input)?;
-        let sig = Signature::from_bytes(&sig)
-            .map_err(|_| CodecError::from("InitializedAssetTx::sig is invalid"))?;
-
-        Ok(InitializedAssetTx { content, sig })
-    }
-}
-
-#[derive(Clone)]
+#[derive(Clone, Encode, Decode)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "std", derive(Debug))]
 pub struct JustifiedAssetTx {
```

Comment:
We no longer need this data type, we should just use `InitializedAssetTx` in place of `JustifiedAssetTx`, and possibly rename it to `AssetTx`.

Reply:
I think it better to keep them separate so that the type system can enforce the workflow.

Reply:
With the way the settlement authorizations work, it's not necessary to store the transaction every step of the workflow, if the mediator authorizes the transaction all of the authorization rules will be satisfied and the mediator will submit the transaction to the validator.

Reply:
I agree with that. But from the MERCAT api design, it is easy for the user of the API to mistakenly pass the output of the first call to the verify function and forgetting to call the justify function.

Reply:
That's a fair point, at the same time casting `InitializedAssetTx` to `JustifiedAssetTx` is trivial. By keeping it this way, on the mesh side we might end up ignoring mediator's returned object and cast the stored `InitializedAssetTx` to `JustifiedAssetTx` to avoid an unnecessary storage; which will make this a bad design.

Reply:
In your scenario, someone need to go out of their way to make that mistake. Since they need to know about the two types and then make a conscious decision bypass the checks to save space. This is not an easy mistake to make ;)

While in my scenario, someone is looking at the input of the validation function and find out that the asset creation function provides that and completely miss the mediator. If that person is not familiar with the business logic, they will most likely make this mistake.

Reply:
I have a feeling we'll end up revisiting this, but I'm happy to be proven wrong ;)